File: PythonCode/OpenCrawlList.py
from WebsiteObject import Website

def openCrawlList():
    
    # Open the crawl list
    f = open("txtFiles/crawlList.txt", "r")
    crawlList = f.read()

    # Split up the crawl list into separate lines - one object is on each line
    crawlList = crawlList.split("\n")

    websiteList = []

    # Create objects for each of the lines
    for i in range(len(crawlList)):
        lineList = crawlList[i].split(": ")
        # Assign each attribute of the website to a variable 
        title = lineList[0]
        keywords = lineList[1].split(", ")
        url = lineList[2]

        # Create the wesbite object from the extracted info
        newWebsite = Website(title, keywords, url)
        websiteList.append(newWebsite)

    return websiteList

if __name__ == "__main__":
    openCrawlList()

File: WebCode/OpenCrawlList.php
<?php

function openCrawlList() {
    // Open the crawl list
    $file = fopen("crawlList.txt", "r");
    $crawlList = fread($file, filesize("crawlList.txt"));

    // Split up the crawl list into separate lines - one object is on each line
    $crawlList = explode("\n", $crawlList);

    $websiteList = [];

    // Create objects for each of the lines
    foreach ($crawlList as $line) {
		// Make the current line a list where each element is separated by where thers a : in the string
        $lineList = explode(": ", $line);
        
        // Assign each attribute of the website to a variable
		$title = $lineList[0];
		
		if (isset($lineList[1])) {
			$keywords = explode(", ", $lineList[1]);
		} else {
			continue;
		}
		
		if (isset($lineList[2])) {
			$url = $lineList[2];
		} else {
			echo "Error for url with: " . $title;
		}
		
		if (isset($lineList[3])) {
			$summary = $lineList[3];
		} else {
			echo "Error for summary with: " . $title;
		}
        
        // Create the website object from the extracted info
        $newWebsite = new Website($title, $keywords, $url, $summary);
        $websiteList[] = $newWebsite;
    }

    fclose($file);
    
    return $websiteList;
}

?>
